--- splash.lua	2013-03-07 13:49:26.621191435 -0500
+++ splash.qs.lua	2013-03-07 12:10:11.303335608 -0500
@@ -1,25 +1,35 @@
 module("luci.controller.splash.splash", package.seeall)
+luci.i18n.loadc("splash")
 
 local uci = luci.model.uci.cursor()
 local util = require "luci.util"
 
 function index()
-	entry({"admin", "services", "splash"}, cbi("splash/splash"), _("Client-Splash"), 90)
+	entry({"admin", "services", "splash"}, cbi("splash/splash"), _("Client-Splash"), 90).i18n = "freifunk"
 	entry({"admin", "services", "splash", "splashtext" }, form("splash/splashtext"), _("Splashtext"), 10)
 
 	local e
 	
 	e = node("splash")
 	e.target = call("action_dispatch")
+	e.i18n = "freifunk"
 
 	node("splash", "activate").target = call("action_activate")
-	node("splash", "splash").target   = template("splash_splash/splash")
+	local uci = luci.model.uci.cursor()
+
+	if uci:get('quickstart', 'options', 'complete') ~= "true" then
+	    node("splash", "splash").target = template("QS/QuickStart")
+	else
+	    node("splash", "splash").target = template("splash_splash/splash")
+	end
+
 	node("splash", "blocked").target  = template("splash/blocked")
 
-	entry({"admin", "status", "splash"}, call("action_status_admin"), _("Client-Splash"))
+	entry({"admin", "status", "splash"}, call("action_status_admin"), _("Client-Splash")).i18n = "freifunk"
 
 	local page  = node("splash", "publicstatus")
 	page.target = call("action_status_public")
+	page.i18n   = "freifunk"
 	page.leaf   = true
 end
 
@@ -54,6 +64,8 @@
 	local ip = luci.http.getenv("REMOTE_ADDR") or "127.0.0.1"
 	local mac = luci.sys.net.ip4mac(ip:match("^[\[::ffff:]*(%d+.%d+%.%d+%.%d+)\]*$"))
 	local blacklisted = false
+  local requesturi = luci.model.uci.cursor():get_bool("luci_splash", "general", "requesturi")
+  local homepage = luci.model.uci.cursor():get("luci_splash", "general", "homepage")
 	if mac and luci.http.formvalue("accept") then
 		uci:foreach("luci_splash", "blacklist",
         	        function(s) if s.mac:lower() == mac or s.mac == mac then blacklisted = true end
@@ -62,7 +74,11 @@
 			luci.http.redirect(luci.dispatcher.build_url("splash" ,"blocked"))
 		else
 			os.execute("luci-splash lease "..mac.." >/dev/null 2>&1")
-			luci.http.redirect(luci.model.uci.cursor():get("freifunk", "community", "homepage"))
+      if requesturi or homepage == "" then
+			  luci.http.redirect(luci.http.getenv("REQUEST_URI"))
+      else
+			  luci.http.redirect(homepage)
+      end
 		end
 	else
 		luci.http.redirect(luci.dispatcher.build_url())
