--- apps_controller.lua	2013-03-12 09:15:04.352750906 -0400
+++ apps_controller.qs.lua	2013-03-12 09:02:56.717520571 -0400
@@ -25,16 +25,19 @@
 require "luci.fs"
 require "commotion_helpers"
 function index()
-    entry({"apps"}, call("load_apps"), "Local Applications", 20).dependent=true
-    entry({"admin","commotion","apps"}, call("admin_load_apps"), "Local Applications", 50).dependent=true
-    --entry({"admin", "commotion", "apps", "list"}, cbi("commotion/apps_cbi")).dependent=true
-    entry({"apps", "add"}, call("add_app")).dependent=true
-    entry({"apps", "add_submit"}, call("action_add")).dependent=true
-    entry({"admin", "commotion", "apps", "edit"}, call("admin_edit_app")).dependent=true
-    entry({"admin", "commotion", "apps", "edit_submit"}, call("action_edit")).dependent=true
-    entry({"admin", "commotion", "apps", "settings"}, call("admin_edit_settings")).dependent=true
-    entry({"admin", "commotion", "apps", "settings_submit"}, call("action_settings")).dependent=true
-    entry({"admin", "commotion", "apps", "judge"}, call("judge_app")).dependent=true
+	local uci = luci.model.uci.cursor()
+   	if uci:get('quickstart', 'options', 'apps') == "true" then
+	   entry({"apps"}, call("load_apps"), "Local Applications", 20).dependent=true
+	   entry({"admin","commotion","apps"}, call("admin_load_apps"), "Local Applications", 50).dependent=true
+	   --entry({"admin", "commotion", "apps", "list"}, cbi("commotion/apps_cbi")).dependent=true
+	   entry({"apps", "add"}, call("add_app")).dependent=true
+	   entry({"apps", "add_submit"}, call("action_add")).dependent=true
+	   entry({"admin", "commotion", "apps", "edit"}, call("admin_edit_app")).dependent=true
+	   entry({"admin", "commotion", "apps", "edit_submit"}, call("action_edit")).dependent=true
+	   entry({"admin", "commotion", "apps", "settings"}, call("admin_edit_settings")).dependent=true
+	   entry({"admin", "commotion", "apps", "settings_submit"}, call("action_settings")).dependent=true
+	   entry({"admin", "commotion", "apps", "judge"}, call("judge_app")).dependent=true
+	end
 end
 
 function judge_app()
